versioned_docs/version-2020fa/lecture3.md

```diff
@@ -39,50 +39,63 @@ necessary, set up a Postman account using your Cornell email address.
 
 #### Code
 
-Create a folder with empty `package.json` and empty `index.js`. Put the
+Create a folder with the following `package.json` and `index.ts`. Put the
 downloaded `service-account.json` in this folder.
 
+Also add `tsconfig.json`, which you can find in [Lecture 2](https://webdev.cornelldti.org/docs/lecture2#tsconfigjson) notes.
+
 Copy the following code into `package.json`.
 
 ```json
 {
   "name": "self-check",
   "version": "0.1.0",
-  "main": "index.js",
+  "main": "index.ts",
   "license": "MIT",
+  "scripts": {
+    "tsc": "tsc -p tsconfig.json",
+    "start": "yarn tsc && node output/index"
+  },
   "dependencies": {
     "body-parser": "^1.18.3",
-    "express": "^4.16.4",
-    "firebase-admin": "^7.2.0"
+    "express": "^4.17.1",
+    "firebase-admin": "^9.2.0"
+  },
+  "devDependencies": {
+    "@types/express": "^4.17.8",
+    "@types/node": "^14.11.5",
+    "typescript": "^4.0.3"
   }
 }
 ```
 
-Copy the following code into `index.js`. Remember to replace `databaseURL` with
+Copy the following code into `index.ts`. Remember to replace `databaseURL` with
 the url of your own. You can find this `databaseURL` in the code snippet in the
 service accounts tab of Firebase.
 
 _Don't worry if you don't understand this code! You'll know what it does and how
 to write similar code by the end of this course!_
 
-```javascript title="index.js"
-const admin = require('firebase-admin');
-const serviceAccount = require('./service-account.json');
-const express = require('express');
-const bodyParser = require('body-parser');
+```typescript title="index.ts"
+import admin from 'firebase-admin';
+import express from 'express';
+import bodyParser from 'body-parser';
+
+// Path to wherever you put your service-account.json
+const serviceAccount = require('../service-account.json');
 
 admin.initializeApp({
   credential: admin.credential.cert(serviceAccount),
-  databaseURL: '[YOUR_OWN_DATABASE_URL]',
+  databaseURL: 'databaseURL',
 });
 
 const db = admin.firestore();
 
 const app = express();
-const port = 8080;
+const port: number = 8080;
```

Comment:
don't have strong opinion on this. depending on the linter config, it will actually warn you on this trivial annotation that can be easily inferred from context.

Reply:
The type was annotated in last semester's notes, but I think it makes sense to infer the type in this case.